diff --git a/au_com_agileware_ewayrecurring.class.php b/au_com_agileware_ewayrecurring.class.php
index 9ae560f..e165805 100644
--- a/au_com_agileware_ewayrecurring.class.php
+++ b/au_com_agileware_ewayrecurring.class.php
@@ -174,7 +174,12 @@ class au_com_agileware_ewayrecurring extends CRM_Core_Payment
    * @return array
    */
   function getPaymentFormFields() {
-    return array();
+    if (Civi::$statics['openedeWayForm'] == 'CRM_Contribute_Form_UpdateBilling') {
+      return parent::getPaymentFormFields();
+    }
+    else {
+      return array();
+    }
   }
 
   /**
@@ -241,6 +246,21 @@ class au_com_agileware_ewayrecurring extends CRM_Core_Payment
         $eWayCustomer['Country'] = $isoCountryCode;
     }
 
+    if (!empty($this->getPaymentFormFields())) {
+
+      $credit_card_name  = $eWayCustomer['FirstName'] . " " . $eWayCustomer['LastName'];
+      $expireYear    = substr ($params['year'], 2, 2);
+      $expireMonth   = sprintf('%02d', (int) $params['month']); // Pad month with zeros
+
+      $eWayCustomer['CardDetails'] = array(
+        'Name' => $credit_card_name,
+        'Number' => $params['credit_card_number'],
+        'ExpiryMonth' => $expireMonth,
+        'ExpiryYear' => $expireYear,
+        'CVN' => $params['cvv2'],
+      );
+    }
+
     return $eWayCustomer;
   }
 
diff --git a/eWAYRecurring.php b/eWAYRecurring.php
index 7a262af..8d53e72 100644
--- a/eWAYRecurring.php
+++ b/eWAYRecurring.php
@@ -489,6 +489,7 @@ function ewayrecurring_civicrm_postProcess ($formName, &$form) {
  * @param $form
  */
 function ewayrecurring_civicrm_preProcess($formName, &$form) {
+  Civi::$statics['openedeWayForm'] = $formName;
   if ($formName == 'CRM_Contribute_Form_Contribution_ThankYou') {
    $paymentProcessor = $form->getVar('_paymentProcessor');
    $paymentProcessor = $paymentProcessor['object'];
